### ISSUE TITLE
Realm for "package data"

### ISSUE THREAD

**Issue by Ms2ger** [whatwg/fetch#400]
https://fetch.spec.whatwg.org/#body-mixin

> package data

… creates a `Blob` without defining its global object.

**Comment by Ms2ger** [comment]
CC @bzbarsky @annevk @domenic 

Assuming that the correct answer is "the global object of the `Body` object we're dealing with"

**Comment by annevk** [comment]
Yeah. As I mentioned elsewhere, IDL really needs to start to define how an object is allocated as an abstract operation and then we should force everyone to go through that whenever they allocate an object of sorts. And then perhaps some convenient functions around that. For instance: Allocate(realm, interfaceToAllocate); AllocateFromThis(this, interfaceToAllocate); AllocateFromCurrent(interfaceToAllocate).

**Comment by domenic** [comment]
I think we should stick with the plan in https://github.com/heycam/webidl/issues/135 as that will be least disruptive. It will only require people to specify the realm when they're doing something asynchronous (i.e. not the sync consequence of a method or constructor call). In that case maybe having a term like "allocate" is good, and then we'd say "allocate an object _x_ using interface _I_ in realm _r_". Otherwise people can say "allocate an object _x_ using interface _I_" or any of their current imprecise phrasings.

**Comment by annevk** [comment]
If that's the plan I guess I don't technically need to change anything here, since we do have a this object.
